src/anyio/_backends/_asyncio.py

```diff
@@ -764,11 +764,15 @@ async def __aexit__(
 
                 self._active = False
                 if self._exceptions:
+                    # The exception that got us here should already have been
+                    # added to self._exceptions so it's ok to break exception
+                    # chaining and avoid adding a "During handling of above..."
+                    # for each nesting level.
                     raise BaseExceptionGroup(
                         "unhandled errors in a TaskGroup", self._exceptions
-                    )
+                    ) from None
                 elif exc_val:
-                    raise exc_val
+                    raise exc_val from None
```

Comment:
I think this needs to restore the cause/traceback/etc in a

```python
try:
    raise exc
finally:
    exc.__cause__ = cause
    exc.__traceback__ = ...
    ...
```

?

Reply:
This seems to defeat the purpose of `raise ... from None`?

Reply:
I'm not sure about if that one is correct.

Reply:
Hmm only way that is hit is if it's a CancelledError as otherwise it would've been added on 733.

It's probably correct to not suppress context in that case as that would lose the frames below the group.
Will revert that.

Using `raise ... from None` doesn't actually remove the `__context__` link though but only sets the `__suppress_context__` flag to avoid printing it in tracebacks by default so any code traversing that for cancelled-errors should work anyway.

Reply:
After some testing it seems like re-raising a caught instance like this doesn't add any extra context and can just be left as is in any case.